### PR TITLE
ACMS-1589: Split acms acms:install command update acms site:install command, install tasks

### PR DESCRIPTION
**Pull request by rajeshreeputra**
Fixes: ACMS-1589
Testing steps: follow from ACMS-1589.

### CODE REVIEW

**Review by chandan-singh7929 — APPROVED** [review]
Looks good to me.

File: src/Cli.php [hunk 1]
```diff
@@ -47,11 +56,18 @@ class Cli {
...
    *   Holds the symfony console output object.
+   * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
+   *   A Symfony container class object.
    */
-  public function __construct(string $project_dir, string $root_dir, OutputInterface $output) {
+  public function __construct(
+    string $project_dir,
+    string $root_dir,
+    OutputInterface $output,
+    ContainerInterface $container) {
```
Comment (apathak18) [inline src/Cli.php:66]: Replace `ContainerInterface $container`  with `Filesystem $filesystem`

File: src/Cli.php [hunk 2]
```diff
@@ -77,12 +93,18 @@ public function printHeadline() :void {
...
+   * Returns an array of information defined in provided file.
+   *
+   * @param string $file_path
+   *   File name to to collect information.
+   *
+   * @return array
+   *   Retuen the file content.
    */
-  public function getAcquiaCmsFile() :array {
+  public function getAcquiaCmsFile(string $file_path) :array {
```
Comment (apathak18) [inline src/Cli.php:104]: replace with public function getAcquiaCmsFile(string $file_path): array {

File: src/Cli.php [hunk 3]
```diff
@@ -94,21 +116,41 @@ public function getAcquiaCmsFile() :array {
...
+   *
+   * @param string $site_uri
+   *   The site uri.
    */
-  public function getInstallerQuestions(string $question_type = NULL) :array {
-    // @todo clean up below logic in ACMS-1589.
-    $fileContent = $this->getAcquiaCmsFile();
-    if ($question_type) {
-      return $fileContent['questions'][$question_type] ?? [];
+  public function getBuildInformtaion(string $site_uri) :array {
```
Comment (apathak18) [inline src/Cli.php:129]: same here

File: src/Helpers/Task/InstallTask.php [hunk 4]
```diff
@@ -166,48 +214,45 @@ public function configure(InputInterface $input, OutputInterface $output, string
...
+    $this->drushCommand->prepare($command)->run();
+    $bundle_modules = $this->buildInformation['modules'] ?? [];
     $modules_list = JsonParser::installPackages($bundle_modules);
     // Get User password from shared factory.
     $password = SharedFactory::getData('password');
     $this->print("User name: admin, User password: $password", 'info');
     $this->print("Enabling modules for the starter-kit:", 'headline');
-    $isDemoContent = $args['keys']['demo_content'] ?? '';
-    if ($isDemoContent == "yes" && ($key = array_search('acquia_cms_starter', $modules_list)) !== FALSE) {
+    $isDemoContent = FALSE;
```
Comment (apathak18) [inline src/Helpers/Task/InstallTask.php:239]: variable names pattern should be consistent throughout the class and methods.

Comment (rajeshreeputra) [reply src/Helpers/Task/InstallTask.php:239]: we will be updating variable name and return type formatting in separate tickets.
cc @vishalkhode1

File: src/Commands/AcmsInstallCommand.php [hunk 5]
```diff
@@ -82,183 +67,32 @@ protected function configure() :void {
...
-      $askQuestion->setAutocompleterValues($question['allowed_values']['options']);
-    }
-    $response = $helper->ask($input, $output, $askQuestion);
-    return ($response === NULL) ? $defaultValue : $response;
+    $install_command = array_merge($install_command, ['--uri=' . $site_uri]);
+    $build_command = array_merge($build_command, $install_command);
+    $build_command = array_merge(['acms:build'], $build_command);
+    $install_command = array_merge([
+      'site:install',
+      '--without-product-info',
```
[the remark this answers is not in the document]
Comment (rajeshreeputra) [reply src/Commands/AcmsInstallCommand.php:89]: proceeding with this

File: src/Commands/AcmsInstallCommand.php [hunk 6]
```diff
@@ -82,183 +67,32 @@ protected function configure() :void {
...
-        $this->acquiaCmsCli->printHeadline();
-        $name = $this->askBundleQuestion($input, $output);
-      }
-      $args['keys'] = $this->askKeysQuestions($input, $output, $name);
-      $this->installTask->configure($input, $output, $name);
-      $this->installTask->run($args);
-      $this->postSiteInstall($name, $output);
+    $build_command = $install_command = [];
+    if ($input->getArgument('name')) {
+      $build_command[] = explode("=", $input->getArgument('name'))[0];
```
[the remark this answers is not in the document]
Comment (rajeshreeputra) [reply src/Commands/AcmsInstallCommand.php:72]: updated

**Review by vishalkhode1 — APPROVED** [review]

**Pull request merged**